test/mail/parsers/rfc_2822_test.exs

```diff
@@ -775,6 +775,121 @@ defmodule Mail.Parsers.RFC2822Test do
              part.headers["content-type"]
   end
 
+  test "parses Windows-1252 encoded filenames" do
+    email = """
+    To: user@example.com
+    From: me@example.com
+    Subject: Test
+    Content-Type: multipart/mixed;
+    	boundary="----=_Part_295474_20544590.1456382229928"
+
+    ------=_Part_295474_20544590.1456382229928
+    Content-Type: text/plain
+
+    This is some text
```

Comment:
Hello,
The charset handler works nicely on the headers, thank you again.
But it would be nice to have it applied to the body parsing too.
An example to reproduce:
```suggestion
    ------=_Part_295474_20544590.1456382229928
    Content-Type: text/plain; charset="Windows-1252"
      Content-Transfer-Encoding: quoted-printable

      fran=E7aise pr=E8s =E0 th=E9=E2tre lumi=E8re
```
Let me know if you need anything.

Reply:
Would that be correctly decoded as "franæaise près à théâtre lumière"?

Reply:
It should be "française près à théâtre lumière".
Thank you.

Reply:
Thanks, I got the character next to ç
I have added the charset handler to also work on body parts where a charset is defined in the content type.

Reply:
Thank you.